kiwmi/desktop/output.c

```diff
@@ -376,8 +388,143 @@ new_output_notify(struct wl_listener *listener, void *data)
     wl_signal_emit(&desktop->events.new_output, output);
 }
 
+void
+output_layout_change_notify(struct wl_listener *listener, void *UNUSED(data))
```

Comment:
Note that this gets called also *every time* we modify the layout from within `output_manager_configure()`. Possible ideas how to deal with it:

- have some state saying whether we’re in `output_manager_configure()` – if yes, return early
  - issue: existence of state similar to previous `kiwmi_output.enable_changed`
- unregister the listener in `o_m_c()`, re-register at the end
  - issue: the listener will soon be used for other purposes (i use it in my PR as well)
- (variation of previous point) unregister the listener, re-register between the 2nd loop and the final `if`s; replace the `set_configuration()` call with a manual emission of the event
  - issue: quite complicated
- keep it as it is and don’t fix what seems to work

Reply:
Should also note that the config can misbehave and change the layout from within the callback, in which case it has to be updated before being sent to clients.

Reply:
> I suppose i can keep nitpicking forever, oh well .

It's okay, I do have a talent for producing extremely nitpickable work

I've opted to just move the contents of that listener into a normal function that's called in the couple of places that it's needed.  Nothing should be changing the output_layout outside of this file so it should work fine (correct me if I'm wrong).  Seemed simpler than having a listener that doesn't actually correspond to the event it's listening to.

Reply:
> Nothing should be changing the output_layout outside of this file so it should work fine

The config can change it. And it seems pretty hard to me to get this right without removing `kiwmi_output:move()`.

(so perhaps leave it be and add a todo? but were i the maintainer, i wouldn’t be happy merging such a thing)

Reply:
Humm, not only the config. If nested kiwmi is requested by its host compositor to resize, it also modifies the output layout without any direct action by us.

Reply:
Good spot and I forgot that the config could move outputs.

I've added output_manager_update calls for both cases (changing the size of nested compositors sets off the mode change listener, so I've put that one in there).  This does make things a little less neat though.

Given that your scene graph stuff also messes around with this, I'd suggest leaving it be for now, then once you've got your scene graph code finalised, you can make an informed decision on what's the neatest way of structuring it.

Reply:
> Given that your scene graph stuff also messes around with this, I'd suggest leaving it be for now, then once you've got your scene graph code finalised, you can make an informed decision on what's the neatest way of structuring it.

Well, scratch that. Looks like I've got a rebase to do.